eq-author/src/App/history/HistoryItem.js

```diff
@@ -45,37 +76,130 @@ const translations = {
   "Questionnaire created": "Questionnaire created",
 };
 
-const formatDate = unformattedDate =>
-  moment(unformattedDate).format("DD/MM/YYYY [at] HH:mm");
+const HistoryButtonGroup = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const SaveButton = styled(Button)`
+  font-size: 0.75em;
+  margin-right: -1em;
+  margin-bottom: 1em;
+`;
+
+const formatDate = date => moment(date).format("DD/MM/YYYY [at] HH:mm");
+
+const RTEWrapper = styled.div`
+  margin: 0 2em 0 1em;
+  padding-bottom: 1em;
+  width: 100%;
+`;
 
 const HistoryItem = ({
+  handleDeleteNote,
+  handleUpdateNote,
   questionnaireTitle,
+  itemId,
   publishStatus,
   userName,
+  currentUser,
+  userId,
   bodyText,
+  type,
   createdAt,
 }) => {
+  const [isEditActive, setIsEditActive] = useState(false);
+  const [noteState, setNoteState] = useState(bodyText);
+  const renderBodyText = () => {
+    if (!isEditActive && bodyText) {
+      return <EventText dangerouslySetInnerHTML={{ __html: bodyText }} />;
+    }
+    if (isEditActive) {
+      return (
+        <StyledGrid>
+          <RTEWrapper>
+            <RichTextEditor
+              id={`update-note-textbox`}
+              name="note"
+              onUpdate={e => setNoteState(e.value)}
+              label=""
+              value={noteState}
+              controls={{
+                heading: true,
+                emphasis: true,
+                list: true,
+                bold: true,
+              }}
+              multiline
+              autoFocus
+            />
+          </RTEWrapper>
+          <ActionButtons horizontal align="right">
+            <SaveButton
+              data-test="save-note-btn"
+              onClick={() =>
+                handleUpdateNote(itemId, noteState).then(() =>
+                  setIsEditActive(false)
+                )
+              }
+            >
+              Save
+            </SaveButton>
+          </ActionButtons>
+        </StyledGrid>
+      );
+    }
+  };
   return (
-    <StyledItem>
-      <QuestionnaireTitle>
-        {questionnaireTitle} -{" "}
-        <strong>{translations[publishStatus] || publishStatus}</strong>
-      </QuestionnaireTitle>
-      <QuestionnaireUserName>
-        {userName} - {formatDate(createdAt)}
-      </QuestionnaireUserName>
-      <BreakLine />
-      {bodyText && <EventText dangerouslySetInnerHTML={{ __html: bodyText }} />}
-    </StyledItem>
+    <EditableItem>
+      {(currentUser.id === userId || currentUser.admin) &&
+        (type === "note" && (
```

Comment:
Can we extract this logic into a `isEditable` variable the conditional rendering here is a little tricky to follow.